### PR TITLE
(650d) Update check your answers page validation

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

Utilise `setFieldErrors` from `FormUtils` to set the validation for the check your answers page.

## Changes in this PR
- Use `FormUtils.setFieldErrors` for the `confirmation` checkbox on check your answers page
- Updated the referrals controller test to reset the response value between each test to avoid unexpected test behaviour.

## Screenshots of UI changes
localhost_3000_referrals_e38891f9-033b-412c-96f1-60b6689a8109_check-answers

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: integration_tests/pages/page.ts
Comment (jsrobertson) [inline integration_tests/pages/page.ts:207]: For a checkbox, the error class is on a form group containing the input.

File: server/controllers/refer/referralsController.test.ts
Comment (jsrobertson) [inline server/controllers/refer/referralsController.test.ts:38]: May not be necessary for `next`

Comment (jsrobertson) [reply server/controllers/refer/referralsController.test.ts:38]: Also open to alternative methods, but this is how I've done it previously.  Set it as a let and then reset it in the `beforeEach`

File: server/controllers/refer/referralsController.test.ts
Comment (jsrobertson) [inline server/controllers/refer/referralsController.test.ts:36]: Just seen you've more or less done the same in #195 @yndajas.  Will wait for that to be merged then rebase.

Comment (yndajas) [reply server/controllers/refer/referralsController.test.ts]: Ah! I see you reset `request` and `next` too - I don't think we ever alter those, but it's probably a good idea in case we do?

File: server/controllers/refer/referralsController.ts
Comment (Gweaton) [inline server/controllers/refer/referralsController.ts:250]: Let's remove "Please" from here - [according to](https://design-system.service.gov.uk/components/error-message/#be-clear-and-concise) the GOV.UK design system it "implies the user has a choice" 😁

**Review by Gweaton — APPROVED** [review]
Looks good to me, just one thing to address before merging

File: server/controllers/refer/referralsController.test.ts
Comment (yndajas) [inline server/controllers/refer/referralsController.test.ts:27]: Could we do this in the other controller test files too?

File: server/controllers/refer/referralsController.test.ts
Comment (yndajas) [inline server/controllers/refer/referralsController.test.ts:269]: Are we happy with the call to `flash` being unmocked?

**Review by yndajas — APPROVED** [review]
Looks good, just a couple of queries

File: server/controllers/refer/referralsController.test.ts
Comment (yndajas) [reply server/controllers/refer/referralsController.test.ts:27]: Commit message here has a typo: `referalls` -> `referrals`

File: server/controllers/refer/referralsController.test.ts
Comment (yndajas) [reply server/controllers/refer/referralsController.test.ts:269]: This was meant in reference to the final commit - does the flash not need mocking there? Perhaps not

File: server/controllers/refer/referralsController.ts
Comment (yndajas) [reply server/controllers/refer/referralsController.ts:250]: "Confirm that" sounds a little harsh though - does it need further revision? "I confirm that"?

File: integration_tests/pages/page.ts
Comment (yndajas) [reply integration_tests/pages/page.ts:207]: Just seen this. Can you mention this in the commit message since there's a change in method functionality (not just use) here?

I suppose the bit above is testing that the field error is present, so this should be okay, unless we can come up with a generic way to test that the error class is present

File: server/controllers/refer/referralsController.test.ts
Comment (yndajas) [reply server/controllers/refer/referralsController.test.ts:269]: (I suppose it makes no difference to the outcome of the action in those conditions, but it's still hit)

File: server/controllers/refer/referralsController.ts
Comment (Gweaton) [reply server/controllers/refer/referralsController.ts:250]: I think it's ok to be direct, I don't think it's too harsh!

File: server/controllers/refer/referralsController.test.ts
Comment (jsrobertson) [reply server/controllers/refer/referralsController.test.ts:269]: `.flash` should already be mocked as a `jest.fn()` as a result of `request = createMock<Request>({ user: { token } })` so it doesn't need it here as well.

**Pull request merged**
